alpha/apps/kaltura/modules/extwidget/actions/thumbnailAction.class.php

```diff
@@ -453,8 +457,8 @@ public function execute()
 		}
 		
 		$nocache = false;
-		if ($secureEntryHelper->shouldDisableCache() || kApiCache::hasExtraFields() ||
-			(!$secureEntryHelper->isKsWidget() && $secureEntryHelper->hasRules(ContextType::THUMBNAIL)))
+		if ($enableCacheValidation && ($secureEntryHelper->shouldDisableCache() || kApiCache::hasExtraFields() ||
```

Comment:
can be optimized ->
$nocache = $enableCacheValidation && ($secureEntryHelper->shouldDisableCache() || kApiCache::hasExtraFields() || !$secureEntryHelper->isKsWidget() && $secureEntryHelper->hasRules(ContextType::THUMBNAIL));